### ISSUE TITLE
when using unplugin-vue/esbuild, an error will occur if there is no dependency on Vite.

### ISSUE THREAD

**Issue by zzfn** [unplugin/unplugin-vue#155]
### Describe the bug

<img width="606" alt="CleanShot 2024-03-10 at 00 44 01@2x" src="https://github.com/unplugin/unplugin-vue/assets/44751363/58029b61-c3fd-4f80-a65a-e9bef52f84e4">

- https://github.com/element-plus/element-plus-icons.git
- Reproducing the issue is possible by downloading the code and running npm run build in the packages/vue directory.

### Reproduction

https://github.com/element-plus/element-plus-icons.git

### System Info

```Shell
System:
    OS: macOS 13.4.1
    CPU: (8) arm64 Apple M1 Pro
    Memory: 79.22 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.10.0 - /opt/homebrew/opt/node@20/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 10.2.3 - /opt/homebrew/opt/node@20/bin/npm
    pnpm: 8.15.4 - /opt/homebrew/bin/pnpm
  Browsers:
    Chrome: 122.0.6261.112
    Chrome Canary: 124.0.6347.0
    Edge: 122.0.2365.80
    Safari: 16.5.1
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sxzz** [comment]
Cannot reproduce. Try to remove `node_modules` and re-install again.

**Comment by sxzz** [comment]
The latest CI is passed. Please refer to https://github.com/element-plus/element-plus-icons/blob/main/.github/workflows/unit-test.yml for usage

https://github.com/element-plus/element-plus-icons

**Comment by zzfn** [comment]
i'm sorry, I didn't make it clear
i encountered this issue while using esbuild separately. 
the steps to reproduce can deleting the "playground" folder. 
i thinking is that vite, a dependency of "playground" is being detected, which is why no errors are being reported.

**Comment by sxzz** [comment]
Please provide a [minimal reproduction](https://stackoverflow.com/help/minimal-reproducible-example) to reopen the issue.
Thanks.

> [Why reproduction is required](https://antfu.me/posts/why-reproductions-are-required)

**Comment by zzfn** [comment]
> Please provide a [minimal reproduction](https://stackoverflow.com/help/minimal-reproducible-example) to reopen the issue. Thanks.
> 
> > [Why reproduction is required](https://antfu.me/posts/why-reproductions-are-required)

it is [reproduction](https://github.com/zzfn/unplugin-vue-issue) (github issue only the comment button)

**Comment by sxzz** [comment]
Thanks for your feedback! Fixed & Released as 5.0.5
